docs/api-reference/core/model.md

```diff
@@ -16,22 +16,94 @@ class.
 
 ## Usage
 
-Create model object by passing shaders, uniforms, and render it by passing updated uniforms.
+### Provide attribute data using Geometry object
+Create model object by passing shaders, uniforms, geometry and render it by passing updated uniforms.
+
 ```js
 // construct the model.
 const model =  new Model(gl, {
   vs: VERTEX_SHADER,
   fs: FRAGMENT_SHADER,
+  uniforms: {uSampler: texture},
   geometry: geometryObject,
-  uniforms: {uSampler: texture}
 })
 
 // and on each frame update any uniforms (typically matrices) and call render.
 model.render({
   uPMatrix: currentProjectionMatrix,
   uMVMatrix: current ModelViewMatrix
 });
+```
+
+### Provide attribute data using Buffer
+When using `Buffer` objects, data remains on GPU and same `Buffer` object can be shared between multiple models.
+
+```js
+// construct the model.
+const model =  new Model(gl, {
+  vs: VERTEX_SHADER,
+  fs: FRAGMENT_SHADER,
+  uniforms: {uSampler: texture},
+  attributes: {
+    attributeName1: bufferObject,
+    attributeName2: {data: dataArray, size: 3, type: GL.FLOAT} // new buffer object will be constructed
```

Comment:
Maybe write `new Float32Array(...)` instead of `dataArray` to emphasize what is expected (a typed array).